diff --git a/CMakeLists.txt b/CMakeLists.txt
index f4bba0000..2b241a000 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -262,7 +262,10 @@ if(UNIX AND NOT APPLE)
     OUTPUT_VARIABLE UNAME_R
     OUTPUT_STRIP_TRAILING_WHITESPACE
   )
-  if(UNAME_R MATCHES ".*[Mm]icrosoft.*" OR UNAME_R MATCHES ".*WSL.*")
+  if(
+    NOT ANDROID
+    AND (UNAME_R MATCHES ".*[Mm]icrosoft.*" OR UNAME_R MATCHES ".*WSL.*")
+  )
     message(
       FATAL_ERROR
       "Building on Windows Subsystem for Linux (WSL) is not supported. If you want to build Mixxx from command line on Windows, you need to use the \"x64 Native Tools Command Prompt for VS 2022\"!"
